source/http_headers.c

```diff
@@ -85,6 +85,29 @@ PyObject *aws_py_http_headers_add(PyObject *self, PyObject *args) {
     Py_RETURN_NONE;
 }
 
+static int s_http_headers_add_pair(PyObject *py_pair, struct aws_http_headers *headers) {
+
+    const char *type_errmsg = "List of (name,value) pairs expected.";
+    if (!PyTuple_Check(py_pair) || PyTuple_Size(py_pair) != 2) {
+        PyErr_SetString(PyExc_TypeError, type_errmsg);
+        return AWS_OP_ERR;
```

Comment:
use python-style errors, or AWS-style errors, but don't mix them like this. It leads to people forgetting to set the "right" kind of thread-local state when errors happen, or assuming that aws_raise_error() was called when it wasn't, and stuff like that

It's not 100% consistent, but I've tried to put "py" in the name of functions that do python-style errors. I'd suggest reworking this like:

~static int s_http_headers_add_pair(...)~ -> `static bool s_py_http_headers_add_pair(...)`

Reply:
Thanks, fixed.